protocols/imap/src/main/java/org/apache/james/imap/processor/NamespaceSupplier.java

```diff
@@ -0,0 +1,78 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.processor;
+
+import java.util.Collection;
+
+import org.apache.james.imap.message.response.NamespaceResponse;
+import org.apache.james.imap.message.response.NamespaceResponse.Namespace;
+import org.apache.james.mailbox.MailboxSession;
+
+import com.google.common.collect.ImmutableList;
+
+public interface NamespaceSupplier {
+    class Default implements NamespaceSupplier {
+        @Override
+        public Collection<Namespace> personalNamespaces(MailboxSession session) {
+            return ImmutableList.of(new NamespaceResponse.Namespace("", session.getPathDelimiter()));
```

Comment:
I know it's a micro-optimization but since we are returning an immutable list can't we cache the list? Maybe using `Suppliers.memoize()`?

Reply:
It's micro optimisation on what is not a critical path, sorry I won't bother: I trust the JIT to do the job!